kong/db/strategies/off/init.lua

```diff
@@ -0,0 +1,150 @@
+local declarative_config = require("kong.db.schema.others.declarative_config")
+
+
+local kong = kong
+local fmt = string.format
+local tostring = tostring
+local tonumber = tonumber
+local encode_base64 = ngx.encode_base64
+local decode_base64 = ngx.decode_base64
+
+
+local off = {}
+
+
+local _mt = {}
+_mt.__index = _mt
+
+
+local function page_for_key(self, key, size, offset)
+  if offset then
+    local token = decode_base64(offset)
+    if not token then
+      return nil, self.errors:invalid_offset(offset, "bad base64 encoding")
+    end
+
+    token = tonumber(token)
+    if not token then
+      return nil, self.errors:invalid_offset(offset, "invalid offset")
+    end
+
+    offset = token
+
+  else
+    offset = 1
+  end
+
+  if not kong.cache then
+    return {}
+  end
+
+  local list = kong.cache:get(key, nil, function()
+    return {}
+  end)
+  if not list then
+    return nil
```

Comment:
Are this branch and this return value correct? Are we missing the second return value from `:get()` being returned from here as well? Should we return `{}`? Maybe it'd help if you could clarify your intentions here (so far I am confused by this function's implementation, it seems).